### ISSUE TITLE
No documentation for status::StatusCode::from_u16/to_u16

### ISSUE THREAD

**Issue by jmatraszek** [hyperium/hyper#942]
I have just spent half an hour trying to figure out how to convert a plain number to a status code and almost ended up implementing this myself... I think documenting this could help some users.

I do not know why those two functions has not been documented. If that is not intended, then I could help and create a pull request (though that would be my first attempt to write documentation in Rust).

**Comment by seanmonstar** [comment]
You can see in the source that they are purposefully hidden from the docs: https://github.com/hyperium/hyper/blob/master/src/status.rs#L232

It may just be wishful thinking, but I wanted to nudge people away from having `StatusCode::from_u16(403)` and whatnot everywhere. That defeats the purpose of having an enum with names, in my opinion.

If the community thinks my nudge just causes pain, I can reconsider.

**Comment by jmatraszek** [comment]
To some point -- I agree with you. In a typical app I would never use `StatusCode::from_u16(200)` instead of `StatusCode::Ok`.

But on the other hand it may be useful in some not-so-typical applications: for example right now I am developing a simple server that will be used to mock APIs, so the status code has nothing to do with the application logic itself, it's just configured in a config file. It is easier to do `StatusCode::from_u16(status_from_config)` than to match by myself.

**Comment by seanmonstar** [comment]
That makes sense. They exist in the first place since the parser only returns a number, so its used as `StatusCode::from_u16(parsed.status)`.

**Comment by jimmycuadra** [comment]
This also confused me a lot. I'm not convinced hiding documentation is the right way to suggest that people avoid a particular API. If it's implemented at all, it should be documented, otherwise it just causes confusion.

My use case is for a trait that abstracts the concept of an HTTP API endpoint, without tying itself directly to Hyper. I was looking for a core/stdlib type so I could do, for example, `type Header: Into<u16>` in the trait, and then a header could be used to construct a Hyper `Header`, but could also be used with some other theoretical HTTP library.

**Comment by 3Hren** [comment]
This is useful for example when writing a proxy server.

**Comment by seanmonstar** [comment]
The ideal resolution here is to have these 2 impls:

```rust
impl From<StatusCode> for u16 {}
impl TryFrom<u16> for StatusCode {
    type Err = InvalidStatusCode;
}
```

The `TryFrom` is because not all `u16`s are valid status codes. Specifically, it must be within `100...599`. The `TryFrom` trait still isn't stable, but the impl could be added behind the `nightly` cargo feature already in hyper.

**Comment by lilydjwg** [comment]
I find `from_u16` is pub now, but `to_u16` is not. I need it to do some logging.

**Comment by seanmonstar** [comment]
@lilydjwg you mean on master? On master there is `u16::from(status)` (or `into()`).

**Comment by lilydjwg** [comment]
@seanmonstar `Into` will consume the StatusCode so I have to clone. I'm inspecting the status code of responses.

**Comment by seanmonstar** [comment]
The status code is Copy, so it should work just fine.

**Comment by lilydjwg** [comment]
No, `u16::from(status)` doesn't work, because there is `impl<T, U> Into<U> for T where U: From<T>` but not the opposite.

This does work:
```
let st: u16 = (*res.status()).into();
```

But it looks awkward. If StatusCode is Copy, I guess it's better for Response.status() to return a copy.
